api/features/views.py

```diff
@@ -110,10 +111,19 @@ def perform_update(self, serializer):
         serializer.save(project_id=self.kwargs.get("project_pk"))
 
     def perform_destroy(self, instance):
-        self._create_feature_delete_audit_log(instance)
+        feature_states = instance.feature_states.filter(
```

Comment:
You might need to wrap this query in `list()` to force it to execute, otherwise, I think  maybe each of the functions below will execute their own query. That might not be true as I think it's the same object being passed into one, then the other but it's probably worth checking / maybe just adding the `list()` to make sure.